cpu/atmega1281/include/cpu_conf.h

```diff
@@ -16,8 +16,8 @@
  * @author          Hinnerk van Bruinehsen <h.v.bruinehsen@fu-berlin.de>
  */
 
-#ifndef __CPU_CONF_H
-#define __CPU_CONF_H
+#ifndef ATMEGA_CPU_CONF_H
```

Comment:
yes, that comment was for the Stellaris headers specifically.
In general, you should name the include guard to what the `#include` line will contain.
if you use `#include "mymodule.h"` to include the file, then its include guard will be `MYMODULE_H_`. If you use `#include "mymodule/config.h"` then you should use an include guard name something like `MYMODULE_CONFIG_H_`

The trailing underscore is up for debate though.